include/pqxx/doc/accessing-results.md

```diff
@@ -56,3 +68,34 @@ This becomes really helpful with the array-indexing operator.  With regular
 C++ iterators you would need ugly expressions like `(*row)[0]` or
 `row->operator[](0)`.  With the iterator types defined by the result and
 row classes you can simply say `row[0]`.
+
+
+Streaming rows
+--------------
+
+There's another way to go through the rows coming out of a query.  It's
+easier and faster, but there are drawbacks.
+
+One, you start getting rows before all the data has come in from the database.
+So if there's an error while you're receiving and processing data then you're
+stuck in the middle.  You might lose your connection to the database when
+you've already started processing the incoming data.
+
+Two, you can't do everything in a streaming query that you can in a regular
+one.  Your query gets wrapped in a PostgreSQL `COPY` command, and `COPY` only
+supports some queries: `SELECT`, `VALUES`, `or an `INSERT`, `UPDATE`, or
+`DELETE` with a `RETURNING` clause.  See the `COPY` documentation here:
+https://www.postgresql.org/docs/current/sql-copy.html
+
+Now for the good news.  Streaming does make it very easy to query data and loop
+over it:
+
+    for (auto [id, name, x, y] :
```

Comment:
You say in other places that there's a risk of losing the database connection mid-stream. I feel like one of the structured bindings should be an std::error_code so that the for-loop can check to see if the network connection was interrupted, and do something about it.

Reply:
There isn't much you _can_ do about it.  You basically lose the connection.  Under those circumstances I think an exception is reasonable.